src/core/contourFeature.js

```diff
@@ -194,9 +198,55 @@ geo.contourFeature = function (arg) {
     if (gridW * gridH > data.length) {
       gridH = Math.floor(data.length) / gridW;
     }
+    /* If we are not using the position values (we are using x0, y0, dx, dy),
+     * and wrapLongitude is turned on, and the position spans 180 degrees,
+     * duplicate one or two columns of points at opposite ends of the map. */
+    usePos = (x0 === null || x0 === undefined || y0 === null ||
+        y0 === undefined || !dx || !dy);
+    if (!usePos && result.wrapLongitude && (x0 < -180 || x0 > 180 ||
+        x0 + dx * (gridW - 1) < -180 || x0 + dx * (gridW - 1) > 180) &&
```

Comment:
> In the general case, this is pretty complicated. For instance, points are easy, but what should lines do? You don't necessarily want to change from drawing a line from 179E, 0N to 181E, 0N to drawing to 179E, 0N to 179W, 0N, as suddenly you've draw a line that is 358 degrees long rather than 2, and that could be very misleading. Similarly, a polygon representing an island at straddling 180 degrees would become the width of the world.

Couldn't agree more. What I was referring to the idea that when we figure out the winding we split the cell (and copy data as necessary) and start it on the other end. It is pretty common use-case in climate packages such as UV-CDAT and NCL.